phoenix-core/src/main/java/org/apache/phoenix/mapreduce/PhoenixMultiViewInputFormat.java

```diff
@@ -0,0 +1,85 @@
+ * limitations under the License.
+ */
+package org.apache.phoenix.mapreduce;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.phoenix.mapreduce.util.DefaultPhoenixMultiViewListProvider;
+import org.apache.phoenix.mapreduce.util.PhoenixConfigurationUtil;
+import org.apache.phoenix.mapreduce.util.PhoenixMultiViewListProvider;
+import org.apache.phoenix.mapreduce.util.DefaultMultiViewSplitStrategy;
+import org.apache.phoenix.mapreduce.util.ViewInfoWritable;
+import org.apache.phoenix.mapreduce.util.MultiViewSplitStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PhoenixMultiViewInputFormat<T extends Writable> extends InputFormat<NullWritable,T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PhoenixMultiViewInputFormat.class);
+
+    public PhoenixMultiViewInputFormat() {
+    }
+
+    @Override public List<InputSplit> getSplits(JobContext context) {
+        List<InputSplit> listOfInputSplit = new ArrayList<>();
+        try {
+            final Configuration configuration = context.getConfiguration();
+            Class<?> defaultViewTtlDeletionInputStrategyClazz = DefaultPhoenixMultiViewListProvider.class;
+            if (configuration.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_INPUT_STRATEGY_CLAZZ) != null) {
+                defaultViewTtlDeletionInputStrategyClazz = Class.forName(
+                        configuration.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_INPUT_STRATEGY_CLAZZ));
+            }
+            PhoenixMultiViewListProvider phoenixMultiViewListProvider =
+                    (PhoenixMultiViewListProvider) defaultViewTtlDeletionInputStrategyClazz.newInstance();
+            List<ViewInfoWritable> viewsWithTTL = phoenixMultiViewListProvider.getPhoenixMultiViewList(configuration);
+
+            Class<?> defaultViewTtlDeletionSplitStrategyClazz = DefaultMultiViewSplitStrategy.class;
+            if (configuration.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_SPLIT_STRATEGY_CLAZZ) != null) {
+                defaultViewTtlDeletionSplitStrategyClazz = Class.forName(
+                        configuration.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_SPLIT_STRATEGY_CLAZZ));
+            }
+            MultiViewSplitStrategy multiViewSplitStrategy =
+                    (MultiViewSplitStrategy) defaultViewTtlDeletionSplitStrategyClazz.newInstance();
+            listOfInputSplit = multiViewSplitStrategy.generateSplits(viewsWithTTL, configuration);
+        } catch (Exception e) {
+            LOGGER.debug(e.getStackTrace().toString());
```

Comment:
Don't swallow the exception and log with error.

Reply:
in this block, it might occur `ClassNotFoundException`, `InstantiationException`, and `IllegalAccessException`; however, the base class `InputFormat` doesn't throw a list above. It has 2 options here.
1. wrapper exception to be either `IOException` or `InterruptedException` and throw it.
2. keep current the implementation, which logs exception message.

Reply:
I think we should wrap as an IOException and rethrow